plaidml/core/core.h

```diff
@@ -339,6 +295,52 @@ class Buffer {
   std::shared_ptr<plaidml_buffer> ptr_;
 };
 
+///
+/// \ingroup core_objects
+/// \class Program
+///
+class Program {
+ public:
+  explicit Program(plaidml_program* ptr) : ptr_(details::make_ptr(ptr)) {}
+
+  ///
+  /// Returns a textual representation of the program.
+  ///
+  std::string str() const { return ffi::str(ffi::call<plaidml_string*>(plaidml_program_repr, as_ptr())); }
+
+  ///
+  /// Compiles a program using the specified target.
+  ///
+  void compile(const std::string& target = "", bool debug = false) {
+    ffi::call_void(plaidml_program_compile, as_ptr(), debug, target.c_str());
+  }
+
+  std::vector<TensorShape> inputs() const {
+    auto shapes = details::make_ptr(ffi::call<plaidml_shapes*>(plaidml_program_get_inputs, as_ptr()));
+    std::vector<TensorShape> ret(shapes->size);
+    for (size_t i = 0; i < shapes->size; i++) {
+      ret[i] = TensorShape(shapes->elts[i]);
+    }
+    return ret;
+  }
+
+  std::vector<TensorShape> outputs() const {
+    auto shapes = details::make_ptr(ffi::call<plaidml_shapes*>(plaidml_program_get_outputs, as_ptr()));
+    std::vector<TensorShape> ret(shapes->size);
+    for (size_t i = 0; i < shapes->size; i++) {
+      ret[i] = TensorShape(shapes->elts[i]);
+    }
+    return ret;
+  }
+
+  Buffer save() const { return Buffer(ffi::call<plaidml_buffer*>(plaidml_program_save, as_ptr())); }
```

Comment:
like the idea of save(). Could we also have a load() function that could consume a previously saved program? 
Would also vote for import/export naming instead of save/load, but that's just a nit

Reply:
Export is good. Could also pass in a 'format' for the export.

Reply:
Downside with `export` is that this is a reserved keyword in c++.